PhpUnit/DefinitionHasMethodCallConstraint.php

```diff
@@ -27,13 +33,17 @@ public function evaluate($other, $description = '', $returnResult = false)
             );
         }
 
-        foreach ($other->getMethodCalls() as $methodCall) {
-            list($method, $arguments) = $methodCall;
+        for ($i = 0; $i < $iMax = count($methodCalls = $other->getMethodCalls()); $i++) {
```

Comment:
This is compact, but I don't think it helps with readability. `$i` should be `$currentIndex`, `$methodCalls` in variable defined outside the loop and `count(...)` might just as well count `$methodCalls`.